wgengine/wgcfg/wgcfg_view.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Code generated by tailscale.com/cmd/cloner; DO NOT EDIT.
+//go:generate go run tailscale.com/cmd/cloner -type=Config,Peer
+
+package wgcfg
+
+import (
+	"encoding/json"
+	"errors"
+
+	"inet.af/netaddr"
+	"tailscale.com/types/key"
+	"tailscale.com/types/views"
+)
+
+// View returns a readonly view of Config.
+func (p *Config) View() ConfigView { return ConfigView{p} }
+
+type ConfigView struct {
```

Comment:
do we need this? I'd prefer to opt-in to which views it generates, rather than it just generating code for everything we need we want a Clone for.

Reply:
I disagree, I think having the ability to defer a `Clone` by providing a `View` is really useful. `View` also relies on `Clone` to provide the `AsStruct` functionality.

As for this exact usage, turns out we can indeed replace our current usage of `Clone` with `View`.
https://github.com/tailscale/tailscale/blob/7f45734663aecae79c8dd86214bbf872ecc3baf8/wgengine/userspace.go#L873

Reply:
https://github.com/tailscale/tailscale/blob/7f45734663aecae79c8dd86214bbf872ecc3baf8/wgengine/userspace.go#L1457-L1467

Reply:
I'm fine with views implying clones, but I'd like use of cloner to not require our religion about views. (Or if this is purely an internal tool, let's say so?)

In any case, I prefer N small tools to 1 megatool that does everything. And I prefer explicitness (generate this view!) over magic.

Reply:
I assumed it was internal because it was in this repo and not a separate one. I'll modify it to add a cmd/viewer

Reply:
Sure. But then we need to update the docs. The package doc only talks about Clone.

And I don't want to see any types where Cloner would've worked but Viewer fails and that prevents Cloner from working.

Reply:
cmd/viewer SGTM. I missed your last sentence.